### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Changes

* Added a pre-commit configuration (based on https://github.com/Ouranosinc/PAVICS-landing/blob/master/.pre-commit-config.yaml)
* Ran hooks over notebooks and source files
* Enabled pre-commit CI checks and monthly pre-commit auto-updates

Needed for PR For new Jupyter env in https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/106.

### CODE REVIEW

File: docs/source/notebooks/opendap.ipynb [hunk 1]
```diff
@@ -26,8 +26,9 @@
    "source": [
     "%matplotlib inline\n",
     "import warnings\n",
-    "import numpy as np\n",
+    "\n",
     "import netCDF4 as nc\n",
+    "import numpy as np\n",
```
Comment (tlvu) [inline docs/source/notebooks/opendap.ipynb:31]: This auto import re-ordering might possibly be problematic in some specific case.  Is there any case where we need some specific import ordering?  Ex: some module monkeypatch another module.

File: docs/source/notebooks/climex.ipynb [hunk 2]
```diff
@@ -101,12 +101,14 @@
...
-    "import xarray as xr\n",
     "import intake\n",
+    "import xarray as xr\n",
     "import xclim\n",
     "from dask.diagnostics import ProgressBar\n",
     "\n",
-    "cat = intake.open_esm_datastore(\"https://pavics.ouranos.ca/catalog/climex.json\")  # TEST_USE_PROD_DATA\n",
+    "cat = intake.open_esm_datastore(\n",
+    "    \"https://pavics.ouranos.ca/catalog/climex.json\"\n",
+    ")  # TEST_USE_PROD_DATA\n",
```
Comment (tlvu) [inline docs/source/notebooks/climex.ipynb:111]: Will break test as well, `# TEST_USE_PROD_DATA` not on same line.

File: docs/source/locale/fr/LC_MESSAGES/arch/data_catalog.po [hunk 3]
```diff
@@ -53,4 +53,3 @@ msgid ""
 " will search for all files that are part of the RCP8.5 experiment and "
 "based on the CRCM4 model."
 msgstr ""
-
```
Comment (tlvu) [inline docs/source/locale/fr/LC_MESSAGES/arch/data_catalog.po:56]: All the `.po` files seem to have an empty last line.  Is this an accident or the format requires it?

File: docs/source/conf.py [hunk 4]
```diff
@@ -144,13 +145,13 @@
 #
 # on_rtd is whether we are on readthedocs.org, this line of code grabbed from docs.readthedocs.org
 # http://docs.readthedocs.org/en/latest/faq.html?highlight=autodoc#how-do-i-change-behavior-for-read-the-docs
-on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
 
-if on_rtd:
-    html_theme = 'default'
+if os.environ.get("READTHEDOCS"):
```
Comment (tlvu) [inline docs/source/conf.py:149]: What?!  We have been using an undefined variable `on_rtd` all this time?!  Did the pre-commit catch that or you did?

**Review by tlvu — CHANGES_REQUESTED** [review]
The issue with `TEST_USE_PROD_DATA` is a deal breaker.

The rest I can live with.

Can we trigger the full set of all the pre-commits, with the same configuration, locally or they can only be triggered on github?  If yes, document somewhere.

We have to be careful when the pre-commits modify data files, ex: `.po` or `.xml` diagram files.  They might or might not be appropriate to be treated as regular text file.

File: docs/source/locale/fr/LC_MESSAGES/arch/data_catalog.po (hunk 3 above)
Comment (Zeitsperre) [reply docs/source/locale/fr/LC_MESSAGES/arch/data_catalog.po:56]: It's a setting for one of the hooks. I should be able to relax it for `.po` files.

File: environment.yml [hunk 5]
```diff
@@ -9,7 +9,6 @@ dependencies:
   - cartopy
   - clisops
   - dask
-#  - descartes
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply environment.yml:12]: I removed it, but I'll put it back.

File: docs/source/notebooks/forecasts.ipynb [hunk 6]
```diff
@@ -83,11 +83,13 @@
...
    "source": [
-    "import xarray as xr\n",
     "import intake\n",
+    "import xarray as xr\n",
     "import xclim\n",
     "\n",
-    "cat = intake.open_esm_datastore(\"https://pavics.ouranos.ca/catalog/forecast.json\")  # TEST_USE_PROD_DATA\n",
+    "cat = intake.open_esm_datastore(\n",
+    "    \"https://pavics.ouranos.ca/catalog/forecast.json\"\n",
+    ")  # TEST_USE_PROD_DATA\n",
```
[the remark this answers is not in the document]
Comment (Zeitsperre) [reply docs/source/notebooks/forecasts.ipynb:92]: Understood. I'll see what I can do about that.

File: docs/source/notebooks/opendap.ipynb (hunk 1 above)
Comment (Zeitsperre) [reply docs/source/notebooks/opendap.ipynb:31]: Valid concern. There used to be an issue with the warning that were raised from CLISOPS on import, but that was dealt with a while ago. xarray-based accessors aren't load-order dependent, so I don't imagine there'd be an issue there either.

File: docs/source/notebooks/climex.ipynb (hunk 2 above)
Comment (Zeitsperre) [reply docs/source/notebooks/climex.ipynb:111]: Gotcha

File: docs/source/locale/fr/LC_MESSAGES/arch/data_catalog.po (hunk 3 above)
Comment (cothel) [reply docs/source/locale/fr/LC_MESSAGES/arch/data_catalog.po:56]: CRCM4 n'a jamais été piloté avec des données RCP8.5
Nos vieilles simulations CRCM$ ont été forcées avec des GCM de CMIP3 forcées par SRES A2

File: setup.py [hunk 7]
```diff
@@ -26,7 +26,7 @@
     author="CRIM/Ouranos",
     author_email="pavics@ouranos.ca",
     description="Scientific gateway for climate data analytics.",
-    long_description=open("README.rst", "rt").read(),
+    long_description=open("README.rst").read(),
```
[the remark this answers is not in the document]
Comment (tlvu) [reply setup.py:29]: it's okay, we can keep it removed.

**Review by huard — APPROVED** [review]
No objection on my end.

File: docs/source/notebooks/climex.ipynb [hunk 8]
```diff
@@ -1866,93 +1868,6 @@
    "cell_type": "code",
    "execution_count": 6,
    "metadata": {},
-   "outputs": [],
-   "source": [
-    "# This will return instantaneously, because xclim is using lazy evaluation whenever possible.\n",
-    "sdii = xclim.atmos.daily_pr_intensity(\n",
-    "    ds.pr.isel(realization=[4, 5], rlat=slice(100, 200), rlon=slice(100, 200)).sel(\n",
```
Comment (tlvu) [inline docs/source/notebooks/climex.ipynb:1873]: Oh this PR is about formatting, normal this section of code is removed?

**Review by tlvu — APPROVED** [review]
Looks good for me now.  I had a comment but it's probably okay.

File: docs/source/notebooks/climex.ipynb (hunk 8 above)
Comment (Zeitsperre) [reply docs/source/notebooks/climex.ipynb:1873]: That's not normal, no. It may have been a strange merge of master ? Will look into this right away.

**Comment by Zeitsperre** [comment]
This PR is waiting on changes from #259 before merging

**Pull request merged**